syntax = "proto3";

service AuthHistory {
    rpc GetAllowListCount(AuthRequest) returns (AllowListCount) {}
    rpc GetBlockListCount(AuthRequest) returns (BlockListCount) {}
    rpc GetCustomAllowListCount(AuthRequest) returns (AllowListCount) {}
    rpc GetCustomBlockListCount(AuthRequest) returns (BlockListCount) {}
    rpc PutAuthEntry(AuthRequest) returns (Empty) {}
}

message AuthRequest {
    string user = 1;
    string password = 2;
    string ip = 3;
    int32 cookie = 4;
    int32 redirect = 5;
    string os = 6;
    string browser = 7;
    string parameters = 8;
    string status = 9;
}

message AllowListCount {
    int32 count = 1;
}

message BlockListCount {
    int32 count = 1;
}

message Empty {

}